### PR TITLE
Simulation of pointer events added for beta Chrome version (closes #903)

### LINKED ISSUE #903: We should support simulation of pointer events for Chrome beta version
New version Chrome will support `pointer events`. Now this behavior in `beta` version.

### PR DESCRIPTION
**Pull request by helen-dikareva**
waiting for tests results

### CODE REVIEW

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
ping @AlexanderMoskovkin @churkin  @VasilyStrelyaev

File: src/client/sandbox/event/simulator.js [hunk 1]
```diff
@@ -369,10 +356,38 @@ export default class EventSimulator {
             Object.defineProperty(pointEvent, 'buttons', {
                 get: () => args.buttons
             });
+        }
+        else {
+            pointerArgs.bubbles    = true;
+            pointerArgs.cancelable = true;
+
+            pointEvent = new window.PointerEvent(pointerEventType, pointerArgs);
```
Comment (churkin) [inline src/client/sandbox/event/simulator.js]: I think we should to use native classes and methods. They are rarely overridden, but it happens.

File: src/client/sandbox/event/simulator.js [hunk 2]
```diff
@@ -298,62 +298,49 @@ export default class EventSimulator {
...
-            }, args);
-
-            pArgs.type       = browserUtils.version > 10 ? 'pointer' + eventShortType : 'MSPointer' +
-                                                                                        eventShortType.charAt(0).toUpperCase() +
-                                                                                        eventShortType.substring(1);
-            pArgs.offsetXArg = args.clientX - elClientPosition.x;
-            pArgs.offsetYArg = args.clientY - elClientPosition.y;
-            pArgs.button     = args.buttons ===
-                               eventUtils.BUTTONS_PARAMETER.noButton ? POINTER_EVENT_BUTTON.noButton : pArgs.button;
+            pointEvent = document.createEvent(browserUtils.isIE10 ? 'MSPointerEvent' : 'PointerEvent');
```
Comment (churkin) [inline src/client/sandbox/event/simulator.js]: use native `document.createEvent`

File: src/client/sandbox/event/simulator.js [hunk 3]
```diff
@@ -227,8 +227,8 @@ export default class EventSimulator {
 
         args.changedTouches = document.createTouchList(args.touch);
```
Comment (churkin) [inline src/client/sandbox/event/simulator.js]: use native document.createTouchList

File: src/client/sandbox/event/simulator.js [hunk 4]
```diff
@@ -369,10 +356,38 @@ export default class EventSimulator {
...
+        var ev            = null;
+        var pointerRegExp = /mouse(down|up|move|over|out)/;
+
+        // NOTE: In IE, submit doesn't work if a click is simulated for some submit button's children (for example,
+        // img, B236676). In addition, if a test is being recorded in IE, the target of a click event is always a
+        // button, not a child, so the child does not receive the click event.
+        if (browserUtils.isIE) {
+            if (args.type === 'click' || args.type === 'mouseup' || args.type === 'mousedown') {
+                if (el.parentNode && domUtils.closest(el.parentNode, 'button')) {
+                    var closestButton = domUtils.closest(el.parentNode, 'button');
```
Comment (churkin) [inline src/client/sandbox/event/simulator.js]: double `domUtils.closest(el.parentNode, 'button');`

File: test/client/fixtures/sandbox/event/event-simulator-test.js [hunk 5]
```diff
@@ -398,6 +398,17 @@ if (!browserUtils.isFirefox) {
     });
 }
 
+if (eventUtils.hasPointerEvents) {
+    test('pointer down', function () {
```
Comment (churkin) [inline test/client/fixtures/sandbox/event/event-simulator-test.js]: I do not understand that the test checks

Comment (helen-dikareva) [inline test/client/fixtures/sandbox/event/event-simulator-test.js]: Test checks that `pointerdown` event are simulated in all browsers which have `pointer` events behavior.

File: src/client/sandbox/event/simulator.js (hunk 3 above)
Comment (helen-dikareva) [inline src/client/sandbox/event/simulator.js]: I'll fix it in https://github.com/DevExpress/testcafe-hammerhead/pull/910

File: Gulpfile.js [hunk 6]
```diff
@@ -98,12 +97,21 @@ var CLIENT_TESTS_BROWSERS = [
     }
 ];
 
+var SAUCELABS_TESTS_BROWSERS = CLIENT_TESTS_BROWSERS;
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: why we need this? Maybe just add 
```js
{
    platform:    'Windows 10',
    browserName: 'chrome',
    version:     'beta'
}
```
to CLIENT_TESTS_BROWSERS?

File: src/client/utils/event.js [hunk 7]
```diff
@@ -46,3 +46,5 @@ export function isObjectEventListener (listener) {
     return typeof listener === 'object' && typeof listener.handleEvent === 'function';
 }
 
+export var hasPointerEvents = !!(window.PointerEvent || window.MSPointerEvent);
```
Comment (AlexanderMoskovkin) [inline src/client/utils/event.js]: The same as @churkin wrote above

> I think we should to use native classes and methods. They are rarely overridden, but it happens.

File: src/client/sandbox/event/simulator.js (hunk 3 above)
Comment (churkin) [inline src/client/sandbox/event/simulator.js]: All methods, ok?

File: test/client/fixtures/sandbox/event/event-simulator-test.js (hunk 5 above)
Comment (churkin) [inline test/client/fixtures/sandbox/event/event-simulator-test.js]: Bad test name for the 'regression' section

**Comment by helen-dikareva** [comment]
Wait for https://github.com/DevExpress/testcafe-hammerhead/pull/919 and fix tests in TestCafe

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

**Pull request merged**